relay/meta.go

```diff
@@ -165,30 +165,30 @@ func (lm *LocalMeta) AdjustWithStartPos(binlogName string, binlogGTID string, en
 		}
 	}
 
-	if (enableGTID && len(binlogGTID) == 0) || (!enableGTID && len(binlogName) == 0) {
-		return false, nil // no meaningful start pos specified
-	}
+	var gset = lm.emptyGSet.Clone()
+	var err error
 
-	if !enableGTID && len(binlogName) > 0 {
-		if !binlog.VerifyFilename(binlogName) {
-			return false, terror.ErrRelayBinlogNameNotValid.Generate(binlogName)
+	if enableGTID {
+		if len(binlogGTID) == 0 {
+			binlogGTID = latestBinlogGTID
 		}
-	}
-	var gset = lm.emptyGSet.Clone()
-	if enableGTID && len(binlogGTID) > 0 {
-		var err error
 		gset, err = gtid.ParserGTID(lm.flavor, binlogGTID)
 		if err != nil {
 			return false, terror.Annotatef(err, "relay-binlog-gtid %s", binlogGTID)
 		}
-	}
-
-	// verified, update them
-	if enableGTID {
-		lm.BinLogName = minCheckpoint.Name
+		lm.BinLogName = ""
```

Comment:
If gtid is enabled in the relay module, the `BinlogName` is useless when pulling relay log, but the relay module still saves a `BinlogName` based on the binlog position each time when flushing relay meta. Shall we unify the usage of `BinlogName` in relay meta?

Reply:
if we get GTID from the upstream master server, we can also get BinlogName at the same time. but if we get GTID from the startup config file, BinlogName is often empty.

But we may still assign BinlogName to meta.

Reply:
so, should we set `BinlogName` here  when enable gtid? @csuzhangxc @amyangfei

Reply:
I prefer to set `BinlogName`

Reply:
update in https://github.com/pingcap/dm/pull/304/commits/e6d94862de9f2676332fc2f0a4f36a261b6e8b35, do you think it's ok @csuzhangxc @amyangfei